imap_processing/idex/tests/test_l1_cdfs.py

```
@@ -0,0 +1,91 @@
+        os.remove(file_name)
+
+
+def test_bad_cdf_attributes(decom_test_data):
+    # Deliberately mess up the attributes to verify that an ISTPError is raised
+    with tempfile.TemporaryDirectory() as tmpdir:
+        del decom_test_data.data["TOF_High"].attrs["DEPEND_1"]
+        with pytest.raises(ISTPError):
+            decom_test_data.write_l1_cdf(version="01", description="", directory=tmpdir)
+
+
+def test_bad_cdf_file_data(decom_test_data):
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # Deliberately mess up the data to verify that an ISTPError is raised
+        bad_data_attrs = {
+            "CATDESC": "Bad_Data",
+            "DEPEND_0": "Epoch",
+            "DISPLAY_TYPE": "no_plot",
+            "FIELDNAM": "Bad_Data",
+            "FILLVAL": "",
+            "FORMAT": "E12.2",
+            "LABLAXIS": "Bad_Data",
+            "UNITS": "",
+            "VALIDMIN": "1",
+            "VALIDMAX": "50",
+            "VAR_TYPE": "support_data",
+            "VAR_NOTES": """How did this data end up in here?
+                            The CDF creation better fail.""",
+        }
+        bad_data_xr = xr.DataArray(
+            name="bad_data",
+            data=np.linspace(1, 50, 50),
+            dims=("bad_data"),
+            attrs=bad_data_attrs,
+        )
+        decom_test_data.data["Bad_data"] = bad_data_xr
+
+        with pytest.raises(ISTPError):
+            decom_test_data.write_l1_cdf(version="01", description="", directory=tmpdir)
+
+
+def test_descriptor_in_file_name(decom_test_data):
+    # Deliberately mess up the data to verify no CDF is created
+    with tempfile.TemporaryDirectory() as tmpdir:
+        file_name = decom_test_data.write_l1_cdf(
+            version="01", description="impact-lab-test001", directory=tmpdir
+        )
+        assert file_name == os.path.join(
+            tmpdir, "imap_idx_l1_20250724_0000_impact-lab-test001_p_v01.cdf"
+        )
```

How did this check that there is no CDF file? out of curiosity

---

Agreed, this should be something like: `assert not file_name.exists()`, but also do we want it to just continue on or should we raise an error/emit a warning if nothing gets created?